### PR TITLE
Add documentation for checkout order creation

### PR DESCRIPTION
**Pull request by lojzatran**
In this documentation I tried to create a quick guide what the customer should do in order to use [commercetools-subscriptions](https://github.com/commercetools/commercetools-subscriptions). The documentation about how commercetools-subscriptions works underneath is NOT part of this PR, it will be added later.

### CODE REVIEW

File: docs/IntegrationGuide.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+## Terminology
+
+- **checkout-order**: an order which customers create during the checkout. It may contain one or more subscriptions.
+- **template-order**: an order which manages a single subscription and is used as a template to create a
+  subscription-order.
+
+## Before you begin
+
+Create all the required custom types and states located in [resources folder](./resources).
```
Comment (ahmetoz) [inline docs/IntegrationGuide.md]:
```suggestion
Create all the required custom types and states located in [resources folder](./resources).  Check [project requirements](./docs/HowToRun.md#commercetools-project-requirements) for more details.
```

File: docs/IntegrationGuide.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+      "quantity": 1,
+      "custom": {
+        "type": {
+          "typeId": "type",
+          "key": "checkout-order-line-item"
+        },
+        "fields": {
+          "cutoffDays": 5,
+          "reminderDays": 5,
+          "subscriptionKey": "",
```
Comment (ahmetoz) [inline docs/IntegrationGuide.md]:
```suggestion
          "subscriptionKey": "UNIQUE_SUBSCRIPTION_KEY",
```

File: docs/IntegrationGuide.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+- **checkout-order**: an order which customers create during the checkout. It may contain one or more subscriptions.
+- **template-order**: an order which manages a single subscription and is used as a template to create a
+  subscription-order.
+
+## Before you begin
+
+Create all the required custom types and states located in [resources folder](./resources).
+
+## Step 1: Create checkout order with subscriptions
```
Comment (ahmetoz) [inline docs/IntegrationGuide.md]:
```suggestion
## Step 1: Create checkout-orders with subscriptions
```

File: docs/IntegrationGuide.md [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    }
+  }
+}
+```
+
+</details>
+
+Checkout order with subscriptions contains following custom fields:
+
+### Checkout order custom fields:
```
Comment (ahmetoz) [inline docs/IntegrationGuide.md:96]:
```suggestion
### checkout-order custom fields:
```

File: docs/IntegrationGuide.md [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+    "fields": {
+      "hasSubscription": true
+    }
+  }
+}
+```
+
+</details>
+
+Checkout order with subscriptions contains following custom fields:
```
Comment (ahmetoz) [inline docs/IntegrationGuide.md:94]:
```suggestion
`checkout-order` with subscriptions contains following custom fields:
```

**Review by ahmetoz — APPROVED** [review]

File: docs/IntegrationGuide.md (hunk 3 above)
Comment (lojzatran) [reply docs/IntegrationGuide.md]: I would use `checkout order`. `checkout` is an adjective describing the noun `order`, same as `template order` or `big order`.  No need to put `-` in between.

File: docs/IntegrationGuide.md (hunk 4 above)
Comment (lojzatran) [reply docs/IntegrationGuide.md:96]: I would use `checkout order`. `checkout` is an adjective describing the noun `order`, same as `template order` or `big order`.  No need to put `-` in between.

File: docs/IntegrationGuide.md (hunk 5 above)
Comment (lojzatran) [reply docs/IntegrationGuide.md:94]: I would use `checkout order`. `checkout` is an adjective describing the noun `order`, same as `template order` or `big order`.  No need to put `-` in between.

**Pull request merged**